src/templates/templateHelpers.ts

```diff
@@ -0,0 +1,10 @@
+/** Wraps a template function and escapes given string arguments. */
+export function createTemplate<T extends (...args: string[]) => string>(
```

Comment:
Not sure I understand the benefit/need for this to be a generic function.

Reply:
The purpose is to preserve type safety / IDE suggestions so if you do something like `const template = createTemplate((one, two) => ....)`
the IDE would enforce `one` and `two` as the string arguments for `template`.

Reply:
Ah, I see, makes sense.